externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetPrimaryKeys.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.flink.operation
+
+import scala.collection.JavaConverters._
+
+import org.apache.commons.lang3.StringUtils
+import org.apache.flink.table.api.{DataTypes, ResultKind}
+import org.apache.flink.table.catalog.Column
+import org.apache.flink.types.Row
+
+import org.apache.kyuubi.engine.flink.result.ResultSet
+import org.apache.kyuubi.operation.meta.ResultSetSchemaConstant._
+import org.apache.kyuubi.session.Session
+
+class GetPrimaryKeys(
+    session: Session,
+    catalogNameOrEmpty: String,
+    schemaNameOrEmpty: String,
+    tableName: String)
+  extends FlinkOperation(session) {
+
+  override protected def runInternal(): Unit = {
+    try {
+      val tableEnv = sessionContext.getExecutionContext.getTableEnvironment
+
+      val catalogName =
+        if (StringUtils.isEmpty(catalogNameOrEmpty)) tableEnv.getCurrentCatalog
+        else catalogNameOrEmpty
+      val catalog = tableEnv.getCatalog(catalogName).get()
+
+      val schemaName =
+        if (StringUtils.isEmpty(schemaNameOrEmpty)) catalog.getDefaultDatabase
+        else schemaNameOrEmpty
```

Comment:
If catalogNameOrEmpty is empty, tableEnv.getCurrentCatalog will be selected. In this case, if schemaNameOrEmpty is empty, schemaName should choose tableEnv.getCurrentDatabase instead of catalog.getDefaultDatabase. WDYT @pan3793

Reply:
good catch, sgtm, @waywtdcc would you please change as @jiaoqingbo suggested?

Reply:
OK, thank you for your review. I have fixed it